engines/director/frame.h

```diff
@@ -88,6 +88,83 @@ struct PaletteInfo {
 	}
 };
 
+struct MainChannels {
+	CastMemberID _actionId;
+	uint16 _transDuration;
+	uint8 _transArea; // 1 - Whole Window, 0 - Changing Area
+	uint8 _transChunkSize;
+	TransitionType _transType;
+	CastMemberID _trans;
+	PaletteInfo _palette;
+	uint8 _tempo;
+
+	CastMemberID _scoreCachedPaletteId;
+
+	CastMemberID _sound1;
+	uint8 _soundType1;
+	CastMemberID _sound2;
+	uint8 _soundType2;
+
+	byte _colorTempo;
+	byte _colorSound1;
+	byte _colorSound2;
+	byte _colorScript;
+	byte _colorTrans;
+
+	uint8 _skipFrameFlag;
+	uint8 _blend;
+
+	MainChannels() {
+		_transDuration = 0;
+		_transType = kTransNone;
+		_transArea = 0;
+		_transChunkSize = 0;
+		_tempo = 0;
+
+		_scoreCachedPaletteId = CastMemberID(0, 0);
+
+		_sound1 = CastMemberID(0, 0);
+		_sound2 = CastMemberID(0, 0);
+		_soundType1 = 0;
+		_soundType2 = 0;
+
+		_actionId = CastMemberID(0, 0);
+		_skipFrameFlag = 0;
+		_blend = 0;
+
+		_colorTempo = 0;
+		_colorSound1 = 0;
+		_colorSound2 = 0;
+		_colorScript = 0;
+		_colorTrans = 0;
+	}
+
+	MainChannels& operator=(const MainChannels& mainChannels) {
```

Comment:
do you need this copy constructor? I believe the default one will work well too.